rehype-trailing-slash.ts

```diff
@@ -0,0 +1,55 @@
+import type { Node } from 'unist';
+import { visit } from 'unist-util-visit';
+
+// Define the type for the element node
+interface ElementNode extends Node {
+  tagName: string;
+  properties: {
+    href?: string;
+  };
+}
+
+const isMailLink = (href: string) => href.startsWith('mailto:');
+
+const isSiteInternalLink = (href: string) => !href.includes('://') || href.includes('nldesignsystem.nl');
+
+const isPageInternalLink = (href: string) => href.startsWith('#');
+
+/**
+ * Append a trailing slash to the pathname portion of the href.
+ * This is done by first building an URL object. With the URL object, the
+ * pathname can be modified separate from the hash or search parts of the url.
+ * Later the origin is removed to preserve the link as written
+ */
+const appendTrailingSlash = (href: string) => {
+  const url = new URL(href, 'https://nldesignsystem.nl');
+  if (!url.pathname.endsWith('/')) {
+    url.pathname += '/';
+  }
+
+  let newHref = url.toString();
+
+  if (href.includes('nldesignsystem.nl') === false) {
+    newHref = newHref.replace(url.origin, '');
+  }
+
+  return newHref;
+};
+
+/**
+ * Rehype plugin looping over every link and appending a trailing slash if it
+ * is an internal link.
+ */
+export default function addTrailingSlash() {
+  return (tree: Node) => {
+    visit(tree, 'element', (node: ElementNode) => {
+      if (node.tagName === 'a' && node.properties && node.properties.href) {
+        const href = node.properties.href;
```

Comment:
Wat als je hier dit doet?
```suggestion
        const url = new URL(node.properties.href, siteURL);
```

met elders:

```js
const siteURL = 'https://nldesignsystem.nl/';
```

Dan kun je bij al je utilities zoals `isPageInternalLink` gebruik maken van de geparsede URL en kun je `url.protocol === 'mailto:'` doen in plaats van `href.startsWith('mailto:')`, waar veel meer edge cases afgevangen worden.